test/deployframework/helpers.go

```diff
@@ -574,11 +576,108 @@ func DeleteAllTestReports(logger logrus.FieldLogger, client meteringclient.Meter
 }
 
 func DeleteAllReportDataSources(logger logrus.FieldLogger, client meteringclient.MeteringV1Interface, namespace string) error {
-	err := client.ReportDataSources(namespace).DeleteCollection(context.TODO(), meta.DeleteOptions{}, meta.ListOptions{})
+	err := client.ReportDataSources(namespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{})
 	if err != nil {
 		return fmt.Errorf("failed to delete all the ReportDataSources in the %s namespace: %v", namespace, err)
 	}
 	logger.Infof("Deleted all of the ReportDataSources in the %s namespace", namespace)
 
 	return nil
 }
+
+// CreateRegistryDeployment is a helper function responsible for building up
+// and creating a local registry Deployment based on the @manifestPath YAML manifest
+// object. In the case where the metering-operator or reporting-operator images have
+// been overrided, we need to do some manipulation of the decoded deployment object
+// to ensure we propagate those custom images when manipulating the CSV images.
+func CreateRegistryDeployment(
+	logger logrus.FieldLogger,
+	client kubernetes.Interface,
+	manifestPath,
+	registryImage,
+	meteringOperatorImage,
+	reportingOperatorImage,
+	namespace string,
+) error {
+	if registryImage == "" {
+		return fmt.Errorf("error: the registry image parameter is empty")
+	}
+
+	var deployment *appsv1.Deployment
+	err := deploy.DecodeYAMLManifestToObject(manifestPath, &deployment)
+	if err != nil {
+		return fmt.Errorf("failed to decode the %s YAML deployment manifest: %v", manifestPath, err)
+	}
+
+	// update the default registry image with the @registryImage provided
+	deployment.Spec.Template.Spec.Containers[0].Image = registryImage
+	deployment.Spec.Template.Spec.InitContainers[0].Image = registryImage
+
+	/*
+		In the case where we're running the testing suite in CI,
+		or if a dev has specified the METERING_OPERATOR_IMAGE_* or
+		REPORTING_OPERATOR_IMAGE_* environment variables, then update
+		the deployment manifest with these values.
+
+		Note: we can somewhat safely make the assumption that the environment
+		variable for the metering-operator is going to be at the first
+		index in that initContainer .Env array, and the reporting-operator
+		is going to be in the second index, but this is a bit gross, so we
+		need to figure out a better way of manipulating this manifest without
+		blindly indexing into this the environment array.
+	*/
+	if meteringOperatorImage != "" {
+		logger.Infof("Overriding the default metering-ansible-operator image with %s", meteringOperatorImage)
+		deployment.Spec.Template.Spec.InitContainers[0].Env[0].Value = meteringOperatorImage
+	}
+	if reportingOperatorImage != "" {
+		logger.Infof("Overriding the default metering-reporting-operator image with %s", reportingOperatorImage)
+		deployment.Spec.Template.Spec.InitContainers[0].Env[1].Value = reportingOperatorImage
+	}
```

Comment:
Building on top of what the above comment suggests, we may be able to just get away with creating a constant variable from the environment variable initContainer array, e.g. `meteringOperatorEnvVarIndex = 0`, to help provide some more context.